admin activity log: wrong user time zone reported #4525

bamboo3250 opened the pull request:
fix #4525

The issue happens only when the user is unregistered and his name is used in the query instead of his google id. Therefore, to solve this, I introduce "course" attribute to the query to indicate the course that unregistered user is belonged to. The query to search for activities of the user will be done via Search page by clicking on "View recent actions".

@kanghj could you review this?

kanghj commented:
While it solves this particular issue, the "course" attribute does not work as advertised. If I search using "course" without an attribute for the user, then the results are the same as if the query was an empty query?

kanghj commented on src/main/java/teammates/ui/controller/AdminActivityLogPageAction.java:
```diff
@@ -118,6 +118,12 @@ private void generateStatusMessage(AdminActivityLogPageData data, List<ActivityL
         if (data.isPersonSpecified()) {
             String targetUserGoogleId = data.getPersonSpecified();
             targetTimeZone = getLocalTimeZoneForRequest(targetUserGoogleId, "");
+            // if cannot find the user or the user is unregistered
```
I think this is an inaccurate comment, there are other reasons why the time zone will be uninitialised

bamboo3250 commented:
The course does not affect the filter but only the timezone shown to admin. Perhaps it's still better to put the course in `role`.

bamboo3250 commented:
@kanghj Now I removed `course` from the query and replaced by using POST action to pass `courseId` from Search Page to Activity Log Page. Please review again.
